packages/ai-jsx/src/batteries/sidekick/platform/large-response-handler.tsx

```diff
@@ -0,0 +1,192 @@
+import * as AI from '../../../index.js';
+import { ConversationMessage, FunctionResponse } from '../../../core/conversation.js';
+
+import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';
+import { getEncoding } from 'js-tiktoken';
+import yaml from 'js-yaml';
+import _ from 'lodash';
+import { UseToolsProps } from '../../use-tools.js';
+import { cohereContext, reranker } from '../../../lib/cohere.js';
+
+export type LengthFunction = ((text: string) => number) | ((text: string) => Promise<number>);
+export interface RedactedFuncionResponseMetadata {
+  isRedacted: true;
+  chunks: string[];
+}
+
+const getEncoder = _.once(() => getEncoding('cl100k_base'));
```

Comment:
I'm arbitrarily using OpenAI's tokenizer here. Trying to tie this back to the actual model being used didn't seem worth it.

Reply:
I guess the dumbest/simplest option to consider would just be to use string length?

Reply:
Hmm, I agree that that's more natural and less opinionated (which is preferred for a library). In practice though I kinda want to use OpenAI's tokenizer for sidekicks.

Specially because we probably want to maintain the parity between chunks from a corpus and chunks from an API (easier to reason about I guess?).


Idea: What if we push `tokenizer`, `lengthFunction`, and such to the context too?

Reply:
To be clear I agree with you that just using OpenAI is the right choice vs. doing largely performative plumbing -- was just throwing out an "even simpler" option for you to (re)consider :)